src/main/java/edu/harvard/iq/dataverse/dataaccess/S3AccessIO.java

```diff
@@ -69,18 +72,49 @@ public S3AccessIO(T dvObject) {
     public S3AccessIO(T dvObject, DataAccessRequest req) {
         super(dvObject, req);
         this.setIsLocalFile(false);
+        
         try {
-            s3 = AmazonS3ClientBuilder.standard().defaultClient();
+            // get a standard client, using the standard way of configuration the credentials, etc.
+            AmazonS3ClientBuilder s3CB = AmazonS3ClientBuilder.standard();
+            // if the admin has set a system property (see below) we use this endpoint URL instead of the standard ones.
+            if (!s3url.isEmpty()) {
+                s3CB.setEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(s3url, s3region));
+            }
+            // some custom S3 implementations require "PathStyleAccess" as they us a path, not a subdomain. default = false
+            s3CB.withPathStyleAccessEnabled(s3pathStyleAccess);
+            // let's build the client :-)
+            this.s3 = s3CB.build();
         } catch (Exception e) {
             throw new AmazonClientException(
-                    "Cannot instantiate a S3 client using; check your AWS credentials and region",
-                    e);
+                        "Cannot instantiate a S3 client using; check your AWS credentials and region",
+                        e);
         }
     }
+    
+    public S3AccessIO(T dvObject, DataAccessRequest req, @NotNull AmazonS3 s3client) {
+        super(dvObject, req);
+        this.setIsLocalFile(false);
+        this.s3 = s3client;
+    }
 
     public static String S3_IDENTIFIER_PREFIX = "s3";
     
     private AmazonS3 s3 = null;
+    /**
+     * Pass in a URL pointing to your S3 compatible storage.
+     * For possible values see https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/client/builder/AwsClientBuilder.EndpointConfiguration.html
+     */
+    private String s3url = System.getProperty("dataverse.files.s3-url", "");
+    /**
+     * Pass in a region to use for SigV4 signing of requests.
+     * Defaults to "dataverse" as it is not relevant for custom S3 implementations.
+     */
+    private String s3region = System.getProperty("dataverse.files.s3-region", "dataverse");
```

Comment:
Hey @matthew-a-dunlap,

the region system property is only used when using a [custom EndpointConfiguration](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/client/builder/AwsClientBuilder.EndpointConfiguration.html), which is only built when a URL is given. Otherwise the AWS client should use the region from `~/.aws/config`. (Although IMHO I cannot test that without proper integration tests, right?)

For most custom S3 services, the region setting is not relevant, thus the default to "dataverse". I can try do dig inside the AWS code to find out how to grab the region from the config. Or we just remove the setting and use a sane default from the [`Regions` AWS enum](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/regions/Regions.html).

Reply:
Ok I confirmed that region config works for AWS S3. I think we'll be ok with the region system property for custom endpoints, though we might want to rename it to make it more clear that it is only needed for custom endpoints. I'll hand this off to our doc/messaging folks to decide on that.

Reply:
I think this can be resolved for now, right?